### PR TITLE
📖 update releasing.md with release branches

### PR DESCRIPTION
**Pull request by tuminoid**
BMO branching proposal was approved in community meeting on 2023-06-21 and BMO release process will have release branches going into future. Update the release process document to reflect this.

Also update CONTRIBUTING.md which details some of the same things.

Fixes: #1283

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -61,22 +62,17 @@ Types and functions not in public APIs are not considered part of the guarantee.
...
-
-We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+We only accept backports of critical bugs, security issues, or bugs without
+easy workarounds, any
 backport MUST not be breaking for either API or behavioral changes.
 We generally do not accept PRs against older release branches.
 
 ## Branches
 
 Baremetal Operator has only the *main* branch currently.
```
Comment (lentzi90) [inline CONTRIBUTING.md:72]: This should be updated I think.
```suggestion
Baremetal Operator has the *main* branch and *release-X* branches.
```

Comment (tuminoid) [reply CONTRIBUTING.md]: Fixed.

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Pull request merged**